### ISSUE TITLE
ERROR {startup} VERSION file '/home/arangodb/dbserver8530/data/VERSION-1' is empty

### ISSUE THREAD

**Issue by Himan000** [arangodb/arangodb#8977]
## My Environment

* __ArangoDB Version__:        3.4.3
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Cluster
* __Deployment Strategy__:   ArangoDB Starter
* __Configuration__:               
* __Infrastructure__:               own
* __Operating System__:        CentOS 7
* __Total RAM in your machine__:        32Gb
* __Disks in use__:        HDD
* __Used Package__:              RedHat .rpm

## Component, Query & Data

__Affected feature__:
server start

## Steps to reproduce

1. The cluster is OK before, but today can't start.
2. start command:
`arangodb --all.query.registry-ttl=3600000 --starter.data-dir=/home/arangodb --server.storage-engine=rocksdb --starter.host=192.168.247.249`

__Problem__:
error as below:
```
_2019-05-13T18:44:03+08:00 |INFO| Starting arangodb version 0.14.0, build ca2ddf9 component=arangodb
2019-05-13T18:44:03+08:00 |INFO| Relaunching service with id '26b8067a' on :8528... component=arangodb
2019-05-13T18:44:03+08:00 |INFO| Using storage engine 'rocksdb' component=arangodb
2019-05-13T18:44:03+08:00 |INFO| ArangoDB Starter listening on 192.168.247.249:8528 (:8528) component=arangodb
2019-05-13T18:44:03+08:00 |INFO| Looking for a running instance of agent on port 8531 component=arangodb
2019-05-13T18:44:03+08:00 |INFO| Starting agent on port 8531 component=arangodb
2019-05-13T18:44:04+08:00 |INFO| Looking for a running instance of dbserver on port 8530 component=arangodb
2019-05-13T18:44:04+08:00 |INFO| Starting dbserver on port 8530 component=arangodb
2019-05-13T18:44:05+08:00 |INFO| agent up and running (version 3.4.3). component=arangodb
2019-05-13T18:44:05+08:00 |INFO| Looking for a running instance of coordinator on port 8529 component=arangodb
2019-05-13T18:44:05+08:00 |INFO| Starting coordinator on port 8529 component=arangodb
2019-05-13T18:44:07+08:00 |INFO| dbserver has terminated quickly, in 2.429390204s (recent failures: 1) component=arangodb
2019-05-13T18:44:07+08:00 |INFO| restarting dbserver component=arangodb
2019-05-13T18:44:07+08:00 |INFO| Looking for a running instance of dbserver on port 8530 component=arangodb
2019-05-13T18:44:07+08:00 |INFO| Starting dbserver on port 8530 component=arangodb
2019-05-13T18:44:09+08:00 |INFO| dbserver has terminated quickly, in 2.034150696s (recent failures: 2) component=arangodb
2019-05-13T18:44:09+08:00 |INFO| ## Start of dbserver log
	2019-05-13T10:44:06Z [70677] WARNING {threads} --server.threads (64) is more than eight times the number of cores (4), this might overload the server
	2019-05-13T10:44:06Z [70677] INFO {authentication} Authentication is turned off, authentication for unix sockets is turned on
	2019-05-13T10:44:06Z [70677] INFO {cluster} Cluster feature is turned on. Agency version: {"server":"arango","version":"3.4.3","license":"community"}, Agency endpoints: http+tcp://192.168.247.250:8531, http+tcp://192.168.247.251:8531, http+tcp://192.168.247.249:
8531, server id: 'PRMR-279664aa-3024-4d23-a45d-c91a9fc27273', internal endpoint / address: tcp://192.168.247.249:8530', advertised endpoint: , role: PRIMARY
	2019-05-13T10:44:06Z [70677] INFO {cluster} using heartbeat interval value '1000 ms' from agency
	2019-05-13T10:44:06Z [70677] ERROR {startup} VERSION file '/home/arangodb/dbserver8530/data/VERSION-1' is empty
	2019-05-13T10:44:06Z [70677] WARNING {startup} in order to automatically fix the VERSION file on startup, please start the server with option `--database.ignore-logfile-errors true`
	2019-05-13T10:44:06Z [70677] FATAL Database '_system' initialization failed (error during startup). Please inspect the logs from the initialization procedure and try starting the server again.
	2019-05-13T10:44:07Z [70837] INFO ArangoDB 3.4.3 [linux] 64bit, using jemalloc, build tags/v3.4.3-0-gda7401391f, VPack 0.1.33, RocksDB 5.16.0, ICU 58.1, V8 5.7.492.77, OpenSSL 1.1.0j  20 Nov 2018
	2019-05-13T10:44:07Z [70837] INFO detected operating system: Linux version 3.10.0-693.el7.x86_64 (builder@kbuilder.dev.centos.org) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-16) (GCC) ) #1 SMP Tue Aug 22 21:09:27 UTC 2017
	2019-05-13T10:44:07Z [70837] INFO {authentication} Jwt secret not specified, generating...
	2019-05-13T10:44:07Z [70837] INFO using storage engine rocksdb
	2019-05-13T10:44:07Z [70837] INFO {cluster} Starting up with role PRIMARY
	2019-05-13T10:44:07Z [70837] INFO {syscall} file-descriptors (nofiles) hard limit is 8192, soft limit is 8192
	2019-05-13T10:44:07Z [70837] WARNING {threads} --server.threads (64) is more than eight times the number of cores (4), this might overload the server
	2019-05-13T10:44:08Z [70837] INFO {authentication} Authentication is turned off, authentication for unix sockets is turned on
	2019-05-13T10:44:08Z [70837] INFO {cluster} Cluster feature is turned on. Agency version: {"server":"arango","version":"3.4.3","license":"community"}, Agency endpoints: http+tcp://192.168.247.250:8531, http+tcp://192.168.247.249:8531, http+tcp://192.168.247.251:
8531, server id: 'PRMR-279664aa-3024-4d23-a45d-c91a9fc27273', internal endpoint / address: tcp://192.168.247.249:8530', advertised endpoint: , role: PRIMARY
	2019-05-13T10:44:08Z [70837] INFO {cluster} using heartbeat interval value '1000 ms' from agency
	2019-05-13T10:44:08Z [70837] ERROR {startup} VERSION file '/home/arangodb/dbserver8530/data/VERSION-1' is empty
	2019-05-13T10:44:08Z [70837] WARNING {startup} in order to automatically fix the VERSION file on startup, please start the server with option `--database.ignore-logfile-errors true`
	2019-05-13T10:44:08Z [70837] FATAL Database '_system' initialization failed (error during startup). Please inspect the logs from the initialization procedure and try starting the server again.
## End of dbserver log component=arangodb
2019-05-13T18:44:09+08:00 |INFO| restarting dbserver component=arangodb
2019-05-13T18:44:09+08:00 |INFO| Looking for a running instance of dbserver on port 8530 component=arangodb
2019-05-13T18:44:09+08:00 |INFO| Starting dbserver on port 8530 component=arangodb
2019-05-13T18:44:11+08:00 |INFO| dbserver has terminated quickly, in 1.971446952s (recent failures: 3) component=arangodb_
```
__Expected result__:
Server start sucessfully.

I have try set option _--database.ignore-logfile-errors true_, and same error occur.
```
_2019-05-13T18:58:03+08:00 |INFO| Starting arangodb version 0.14.0, build ca2ddf9 component=arangodb
2019-05-13T18:58:03+08:00 |INFO| Relaunching service with id '26b8067a' on :8528... component=arangodb
2019-05-13T18:58:03+08:00 |INFO| Using storage engine 'rocksdb' component=arangodb
2019-05-13T18:58:03+08:00 |INFO| ArangoDB Starter listening on 192.168.247.249:8528 (:8528) component=arangodb
2019-05-13T18:58:03+08:00 |INFO| Looking for a running instance of agent on port 8531 component=arangodb
2019-05-13T18:58:03+08:00 |INFO| Starting agent on port 8531 component=arangodb
2019-05-13T18:58:04+08:00 |INFO| Looking for a running instance of dbserver on port 8530 component=arangodb
2019-05-13T18:58:04+08:00 |INFO| Starting dbserver on port 8530 component=arangodb
2019-05-13T18:58:04+08:00 |INFO| dbserver has terminated quickly, in 246.574871ms (recent failures: 1) component=arangodb
2019-05-13T18:58:04+08:00 |INFO| restarting dbserver component=arangodb
2019-05-13T18:58:04+08:00 |INFO| Looking for a running instance of dbserver on port 8530 component=arangodb
2019-05-13T18:58:04+08:00 |INFO| Starting dbserver on port 8530 component=arangodb
2019-05-13T18:58:05+08:00 |INFO| dbserver has terminated quickly, in 258.552161ms (recent failures: 2) component=arangodb
2019-05-13T18:58:05+08:00 |INFO| ## Start of dbserver log
	2019-05-13T10:44:13Z [71234] INFO {authentication} Authentication is turned off, authentication for unix sockets is turned on
	2019-05-13T10:44:14Z [71234] INFO {cluster} Cluster feature is turned on. Agency version: {"server":"arango","version":"3.4.3","license":"community"}, Agency endpoints: http+tcp://192.168.247.250:8531, http+tcp://192.168.247.249:8531, http+tcp://192.168.247.251:
8531, server id: 'PRMR-279664aa-3024-4d23-a45d-c91a9fc27273', internal endpoint / address: tcp://192.168.247.249:8530', advertised endpoint: , role: PRIMARY
	2019-05-13T10:44:14Z [71234] INFO {cluster} using heartbeat interval value '1000 ms' from agency
	2019-05-13T10:44:14Z [71234] ERROR {startup} VERSION file '/home/arangodb/dbserver8530/data/VERSION-1' is empty
	2019-05-13T10:44:14Z [71234] WARNING {startup} in order to automatically fix the VERSION file on startup, please start the server with option `--database.ignore-logfile-errors true`
	2019-05-13T10:44:14Z [71234] FATAL Database '_system' initialization failed (error during startup). Please inspect the logs from the initialization procedure and try starting the server again.
	2019-05-13T10:44:15Z [71366] INFO ArangoDB 3.4.3 [linux] 64bit, using jemalloc, build tags/v3.4.3-0-gda7401391f, VPack 0.1.33, RocksDB 5.16.0, ICU 58.1, V8 5.7.492.77, OpenSSL 1.1.0j  20 Nov 2018
	2019-05-13T10:44:15Z [71366] INFO detected operating system: Linux version 3.10.0-693.el7.x86_64 (builder@kbuilder.dev.centos.org) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-16) (GCC) ) #1 SMP Tue Aug 22 21:09:27 UTC 2017
	2019-05-13T10:44:15Z [71366] INFO {authentication} Jwt secret not specified, generating...
	2019-05-13T10:44:15Z [71366] INFO using storage engine rocksdb
	2019-05-13T10:44:15Z [71366] INFO {cluster} Starting up with role PRIMARY
	2019-05-13T10:44:15Z [71366] INFO {syscall} file-descriptors (nofiles) hard limit is 8192, soft limit is 8192
	2019-05-13T10:44:15Z [71366] WARNING {threads} --server.threads (64) is more than eight times the number of cores (4), this might overload the server
	2019-05-13T10:44:15Z [71366] INFO {authentication} Authentication is turned off, authentication for unix sockets is turned on
	2019-05-13T10:44:15Z [71366] INFO {cluster} Cluster feature is turned on. Agency version: {"server":"arango","version":"3.4.3","license":"community"}, Agency endpoints: http+tcp://192.168.247.250:8531, http+tcp://192.168.247.249:8531, http+tcp://192.168.247.251:
8531, server id: 'PRMR-279664aa-3024-4d23-a45d-c91a9fc27273', internal endpoint / address: tcp://192.168.247.249:8530', advertised endpoint: , role: PRIMARY
	2019-05-13T10:44:15Z [71366] INFO {cluster} using heartbeat interval value '1000 ms' from agency
	2019-05-13T10:44:15Z [71366] ERROR {startup} VERSION file '/home/arangodb/dbserver8530/data/VERSION-1' is empty
	2019-05-13T10:44:15Z [71366] WARNING {startup} in order to automatically fix the VERSION file on startup, please start the server with option `--database.ignore-logfile-errors true`
	2019-05-13T10:44:15Z [71366] FATAL Database '_system' initialization failed (error during startup). Please inspect the logs from the initialization procedure and try starting the server again.
	2019-05-13T10:44:16Z [71497] INFO ArangoDB 3.4.3 [linux] 64bit, using jemalloc, build tags/v3.4.3-0-gda7401391f, VPack 0.1.33, RocksDB 5.16.0, ICU 58.1, V8 5.7.492.77, OpenSSL 1.1.0j  20 Nov 2018
## End of dbserver log component=arangodb
2019-05-13T18:58:05+08:00 |INFO| restarting dbserver component=arangodb
2019-05-13T18:58:05+08:00 |INFO| Looking for a running instance of dbserver on port 8530 component=arangodb
2019-05-13T18:58:05+08:00 |INFO| Starting dbserver on port 8530 component=arangodb
2019-05-13T18:58:05+08:00 |INFO| agent up and running (version 3.4.3). component=arangodb
2019-05-13T18:58:05+08:00 |INFO| dbserver has terminated quickly, in 246.776568ms (recent failures: 3) component=arangodb_
```

Could anyone help? Thanks

**Comment by maxkernbach** [comment]
Hi @Himan000,

As reported in the logs, this startup error is caused by an empty **VERSION** file. I was able to reproduce this state by emptying a **VERSION** file and restarting one starter process.

The printed error message is misleading and already fixed in the upcoming minor release 3.4.6.

> * fixed internal issue 3893: correct wrong option name `--database.ignore-logfile-errors`
> in error message and correct it to `--database.ignore-datafile-errors`

Therefore you should be able to automatically fix the VERSION file on startup by specifying the option `--all.database.ignore-datafile-errors=true`. This command is passing the `--database.ignore-datafile-errors` to all database servers of your cluster (see [Other database options](https://www.arangodb.com/docs/stable/programs-starter-options.html#other-database-options)).

Could you please try to restart the `arangodb` process by adding `--all.database.ignore-datafile-errors=true`.

E.g. your full startup command:

```
arangodb --all.query.registry-ttl=3600000 --starter.data-dir=/home/arangodb --server.storage-engine=rocksdb --starter.host=192.168.247.249 --all.database.ignore-datafile-errors=true
```

**Comment by Himan000** [comment]
Hi @maxkernbach ,
Great, it works after setting the option you provided. Thanks a lot. But I don't know why the  VERSION file is empty suddenly. Never mind, I don't need to care much about it. Thanks.
